tests/cli/deployment/test_deployment_run.py

```diff
@@ -0,0 +1,212 @@
+import dateparser
+import pendulum
+import pytest
+
+import prefect
+from prefect.testing.cli import invoke_and_assert
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+
+@pytest.fixture
+async def deployment_name(deployment, orion_client):
+    flow = await orion_client.read_flow(deployment.flow_id)
+    return f"{flow.name}/{deployment.name}"
+
+
+@pytest.fixture
+def frozen_now(monkeypatch):
+    now = pendulum.now()
+    monkeypatch.setattr("pendulum.now", lambda *_: now)
+    yield now
+
+
+def test_both_start_in_and_start_at_raises():
+    invoke_and_assert(
+        command=["deployment", "run", "--start-in", "foo", "--start-at", "bar"],
+        expected_code=1,
+        expected_output="Expected optional start_in field or start_at field but not both.",
+    )
+
+
+@pytest.mark.parametrize(
+    "start_at,expected_start_time",
+    [
+        ("12-1-22 5pm", "2022-12-01 17:00:00"),
+        ("1/1/30", "2030-01-01 00:00:00"),
+        ("13/1/20 13:31", "2020-01-13 13:31:00"),
+        ("9pm December 31st 2022", "2022-12-31 21:00:00"),
+        ("January 2nd 2023", "2023-01-02 00:00:00"),
+    ],
+)
+async def test_start_at_option_schedules_flow_run_in_future(
+    deployment_name: str,
+    orion_client: prefect.OrionClient,
+    start_at: str,
+    expected_start_time,
+):
+
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            start_at,
+        ],
+        expected_output_contains=["Scheduled start time:", expected_start_time],
+    )
+
+    flow_runs = await orion_client.read_flow_runs()
+    assert len(flow_runs) == 1
+    flow_run = flow_runs[0]
+
+    assert flow_run.state.is_scheduled()
+
+
+@pytest.mark.parametrize(
+    "start_in,expected_duration",
+    [
+        ("10 minutes", pendulum.duration(minutes=10)),
+        ("5 days", pendulum.duration(days=5)),
+        ("3 seconds", pendulum.duration(seconds=3)),
+        (None, pendulum.duration(seconds=0)),
+        ("1 year and 3 months", pendulum.duration(years=1, months=3)),
+        ("2 weeks & 1 day", pendulum.duration(weeks=2, days=1)),
+        ("27 hours + 4 mins", pendulum.duration(days=1, hours=3, minutes=4)),
+    ],
+)
+async def test_start_in_option_schedules_flow_run_in_future(
+    deployment_name: str,
+    frozen_now,
+    orion_client: prefect.OrionClient,
+    start_in: str,
+    expected_duration,
+):
+    expected_start_time = frozen_now + expected_duration
+    expected_display = expected_start_time.in_tz(
+        pendulum.tz.local_timezone()
+    ).to_datetime_string()
+
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-in",
+            start_in,
+        ],
+        expected_output_contains=f"Scheduled start time: {expected_display}",
```

Comment:
It seems like this part could be covered by `test_start_at_displays_scheduled_start_time`